doc/sphinx-guides/source/api/external-tools.rst

```diff
@@ -92,7 +92,7 @@ Terminology
 
     scope                        Whether the external tool appears and operates at the **file** level or the **dataset** level. Note that a file level tool much also specify the type of file it operates on (see "contentType" below).
 
-    types                        Whether the external tool is an **explore** tool, a **preview** tool, a **configure** tool or any combination of these (multiple types are supported for a single tool). Configure tools require an API token because they make changes to data files (files within datasets). Configure tools are currently not supported at the dataset level. The older "type" keyword that allows you to pass a single type as a string is deprecated but still supported.
+    types                        Whether the external tool is an **explore** tool, a **preview** tool, a **query** tool, a **configure** tool or any combination of these (multiple types are supported for a single tool). Configure tools require an API token because they make changes to data files (files within datasets). Configure tools are currently not supported at the dataset level. The older "type" keyword that allows you to pass a single type as a string is deprecated but still supported.
```

Comment:
Regarding multiple types... if you create a tool with both preview and query types like this:

```
{
  "displayName": "Ask Dataverse",
  "description": "A tool that allows you to ask questions about Dataverse tables.",
  "toolName": "askDV",
  "scope": "file",
  "types": [
    "query",
    "preview"
  ],
  "toolUrl": "https://askdataverse.shinyapps.io/askdata",
  "contentType": "text/tab-separated-values",
  "httpMethod":"GET",
  "toolParameters": {
    "queryParameters": [
      {
        "fileid": "{fileId}"
      },
      {
        "datasetPid": "{datasetPid}"
      },
      {
        "siteUrl": "{siteUrl}"
      }
    ]
  }
}
```

... it shows up under the "Preview" tab and is listed twice:

![Screen Shot 2023-07-26 at 2 45 28 PM](https://github.com/IQSS/dataverse/assets/21006/5c46b042-ccf9-46cf-a7a3-fe788eca9b47)

I'm not sure what behavior we want but this seems a bit strange. It's like the Preview tab wins. Maybe we could just document this behavior.

The fact that the tool is listed twice feels like a bug to me. 🐞

Reply:
Updated so that if the tool has multiple types it will only be listed once